### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by gogonzo**
part of #937

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -5,6 +5,13 @@
 * `data` argument in `init` now accepts `teal_data` and `teal_data_module`.
 * Added `landing_popup_module` function which creates a module that will display a popup when the app starts. The popup will block access to the app until it is dismissed.
 * Filter state snapshots can now be uploaded from file. See `?snapshot`.
+* Added `as_tdata` function to facilitate migration of modules to the new `teal_data` class.
+
+### Breaking changes
+
+* `tdata` has been deprecated and replaced with `teal_data`. Support for `tdata` as `data` argument in the `module(server)` will be removed in the next release.
+* `module(ui)` argument no longer accepts `data` and `datasets` arguments. Data dependent logic should be set in the `server` function.
```
Comment (chlebowa) [inline NEWS.md:13]:
```suggestion
* `module(ui)` argument no longer accepts `data` and `datasets` arguments. All data dependent logic should be set in the `server` function.
```

File: NEWS.md [hunk 2]
```diff
@@ -5,6 +5,13 @@
...
+* `module(server)` argument deprecated a `datasets` argument. `teal_module`'s `server` should accept `data` (`teal_data`) instead.
```
Comment (chlebowa) [inline NEWS.md:14]:
```suggestion
* `module(server)` argument deprecated `datasets` argument. `teal_module`s' `server` functions should accept `data` (`teal_data`) instead.
```

File: NEWS.md [hunk 3]
```diff
@@ -5,6 +5,13 @@
 * `data` argument in `init` now accepts `teal_data` and `teal_data_module`.
 * Added `landing_popup_module` function which creates a module that will display a popup when the app starts. The popup will block access to the app until it is dismissed.
 * Filter state snapshots can now be uploaded from file. See `?snapshot`.
+* Added `as_tdata` function to facilitate migration of modules to the new `teal_data` class.
+
+### Breaking changes
+
+* `tdata` has been deprecated and replaced with `teal_data`. Support for `tdata` as `data` argument in the `module(server)` will be removed in the next release.
```
Comment (chlebowa) [inline NEWS.md:12]:
```suggestion
* `tdata` has been deprecated and replaced with `teal_data`. Support for `tdata` passed to the `data` argument in `module(server)` will be removed in the next release.
```

**Review by kartikeyakirar — APPROVED** [review]

File: DESCRIPTION [hunk 4]
```diff
@@ -31,8 +31,8 @@ BugReports: https://github.com/insightsengineering/teal/issues
 Depends:
     R (>= 4.0),
     shiny (>= 1.7.0),
-    teal.data (>= 0.3.0.9011),
-    teal.slice (>= 0.4.0.9023),
+    teal.data (>= 0.3.0.9017),
+    teal.slice (>= 0.4.0.9027),
     teal.transform (>= 0.4.0.9007)
```
Comment (vedhav) [inline DESCRIPTION:36]: Waiting for teal.transform vbump. The version `0.4.0.9010` has refactor changes. But, I'm not sure about the teal.code version over there at teal.transform.

**Pull request merged**